oracle/blockeligibilityvalidator.go

```diff
@@ -15,42 +15,41 @@ type BlockEligibilityValidator struct {
 	activationDb   ActivationDb
 	beaconProvider *EpochBeaconProvider
 	validateVRF    VRFValidationFunction
+	log            log.Log
 }
 
 func NewBlockEligibilityValidator(committeeSize int32, layersPerEpoch uint16, activationDb ActivationDb,
-	beaconProvider *EpochBeaconProvider, validateVRF VRFValidationFunction) *BlockEligibilityValidator {
+	beaconProvider *EpochBeaconProvider, validateVRF VRFValidationFunction, log log.Log) *BlockEligibilityValidator {
 
 	return &BlockEligibilityValidator{
 		committeeSize:  committeeSize,
 		layersPerEpoch: layersPerEpoch,
 		activationDb:   activationDb,
 		beaconProvider: beaconProvider,
 		validateVRF:    validateVRF,
+		log:            log,
 	}
 }
 
 func (v BlockEligibilityValidator) BlockEligible(block *types.Block) (bool, error) {
 	epochNumber := block.LayerIndex.GetEpoch(v.layersPerEpoch)
 
-	atx, err := v.activationDb.GetAtx(block.ATXID)
-	if err != nil {
-		log.Error("getting ATX failed: %v", err)
-		return false, err
-	}
-
-	if err := atx.Validate(); err != nil {
-		log.Error("ATX is invalid: %v", err)
-		return false, err
-	}
-	if atxEpochNumber := atx.LayerIdx.GetEpoch(v.layersPerEpoch); epochNumber != atxEpochNumber {
-		log.Error("ATX epoch (%d) doesn't match layer ID epoch (%d)", atxEpochNumber, epochNumber)
-		return false, fmt.Errorf("activation epoch (%d) mismatch with layer epoch (%d)", atxEpochNumber,
-			epochNumber)
+	// need to get active set size from previous epoch
+	activeSetSize := uint32(GenesisActiveSetSize)
+	if !epochNumber.IsGenesis() {
+		atx, err := v.getValidATX(epochNumber, block)
+		if err != nil {
+			return false, err
+		}
+		activeSetSize = atx.ActiveSetSize
+		if epochNumber.IsGenesis() && activeSetSize < GenesisActiveSetSize {
```

Comment:
epochNumber.IsGenesis() is already checked in the outer. Apparently we will never pass this if (are you missing a test case?)

Reply:
@noamnelke please address this

Reply:
good catch, this `if` can just be removed, I think it was added when we tried to catch a weird case and wasn't cleaned up.